attest/api/proto/attest.proto

```diff
@@ -48,3 +48,20 @@ message Message {
         /// for use in the enclave.
         bytes data = 3;
 }
+
+/// An AEAD message with an explicit nonce.
+///
+/// This message is technically compatible with [`Message`], but exists to
```

Comment:
What does it mean that a `NonceMessage` is compatible with `Message`? And why would we want the generated code to not use `Message` if it's compatible?

Reply:
It's theoretically possible for a gRPC participant to send a `Message` on one side, and have it get decoded as a `NonceMessage` on the other, because the all fields are optional, and a field with a defaulted value is omitted.

So a `NonceMessage` with `nonce = 0` looks identical to a `Message` when encoded.

The reason to split the types up is so that you know one of these should have a nonce and one of them shouldn't _in the code_, even if you can't tell from the bytes.

Reply:
Ah thanks for explaining. I totally agree that we should have explicit type for `NonceMessage` to ensure its bytes aren't confused for a `Message`.